agent/agentlocal/agent_local_test.go

```diff
@@ -109,3 +119,130 @@ func TestServerStatus(t *testing.T) {
+		t.Helper()
+		agentInfo := []*agentlocalpb.AgentInfo{{
+			AgentId:   "/agent_id/00000000-0000-4000-8000-000000000002",
+			AgentType: inventorypb.AgentType_NODE_EXPORTER,
+			Status:    inventorypb.AgentStatus_RUNNING,
+		}}
+		var supervisor mockSupervisor
+		supervisor.Test(t)
+		supervisor.On("AgentsList").Return(agentInfo)
+		agentLogs := make(map[string][]string)
+		agentLogs[inventorypb.AgentType_NODE_EXPORTER.String()] = []string{
+			"logs1",
+			"logs2",
+		}
+		supervisor.On("AgentsLogs").Return(agentLogs)
+		var client mockClient
+		client.Test(t)
+		client.On("GetServerConnectMetadata").Return(&agentpb.ServerConnectMetadata{
+			AgentRunsOnNodeID: "/node_id/00000000-0000-4000-8000-000000000003",
+			ServerVersion:     "2.0.0-dev",
+		})
+		cfg := &config.Config{
+			ID: "/agent_id/00000000-0000-4000-8000-000000000001",
+			Server: config.Server{
+				Address:  "127.0.0.1:8443",
+				Username: "username",
+				Password: "password",
+			},
+		}
+		return agentInfo, &supervisor, &client, cfg
+	}
+
+	t.Run("test zip file", func(t *testing.T) {
+		_, supervisor, client, cfg := setup(t)
+		defer supervisor.AssertExpectations(t)
+		defer client.AssertExpectations(t)
+		ringLog := storelogs.New(10)
+		s := NewServer(cfg, supervisor, client, "/some/dir/pmm-agent.yaml", ringLog)
+		_, err := s.Status(context.Background(), &agentlocalpb.StatusRequest{GetNetworkInfo: false})
+		require.NoError(t, err)
+
+		rec := httptest.NewRecorder()
+		req := httptest.NewRequest("GET", "/logs.zip", nil)
+		s.Zip(rec, req)
+		existFile, err := ioutil.ReadAll(rec.Body)
+		require.NoError(t, err)
+
+		expectedFile, err := generateTestZip(s)
+		require.NoError(t, err)
+		bufExp := bytes.NewReader(expectedFile)
+		bufExs := bytes.NewReader(existFile)
+
+		zipExp, err := zip.NewReader(bufExp, bufExp.Size())
+		require.NoError(t, err)
+		zipExs, err := zip.NewReader(bufExp, bufExs.Size())
+		require.NoError(t, err)
+
+		for i, ex := range zipExp.File {
+			assert.Equal(t, ex.Name, zipExs.File[i].Name)
+			deepCompare(t, ex, zipExs.File[i])
+		}
+		require.NoError(t, err)
+		assert.Equal(t, expectedFile, existFile)
+	})
+}
+
+// generateTestZip generate test zip file.
+func generateTestZip(s *Server) ([]byte, error) {
+	agentLogs := make(map[string][]string)
+	agentLogs[inventorypb.AgentType_NODE_EXPORTER.String()] = []string{
+		"logs1",
+		"logs2",
+	}
+	buf := &bytes.Buffer{}
+	writer := zip.NewWriter(buf)
+	b := &bytes.Buffer{}
+	for _, serverLog := range s.ringLogs.GetLogs() {
+		_, err := b.WriteString(serverLog)
+		if err != nil {
+			return nil, err
+		}
+	}
+	addData(writer, "pmm-agent.txt", b.Bytes())
+
+	for id, logs := range agentLogs {
+		b := &bytes.Buffer{}
+		for _, l := range logs {
+			_, err := b.WriteString(l + "\n")
+			if err != nil {
+				return nil, err
+			}
+		}
+		addData(writer, fmt.Sprintf("%s.txt", id), b.Bytes())
+	}
+	err := writer.Close()
+	if err != nil {
+		return nil, err
+	}
+	return buf.Bytes(), nil
+}
```

Comment:
this code is a copy-paste of the method you test, which means that it will pass anyway.
Can we make a golden file, include it in the repository and compare the method result with that?
If it's not possible because of the body of files being dynamic, can we check that the zip file is correct and contains the files we expected and that body of files is not empty?